generics/services/users.js

```diff
@@ -6,350 +6,365 @@
  */
 
 //dependencies
-const request = require('request');
+const request = require("request");
 const userServiceUrl = process.env.USER_SERVICE_URL;
+const env = process.env.APPLICATION_ENV;
+const jwt = require("jsonwebtoken");
 
-const profile = function ( token,userId = "" ) {
-    return new Promise(async (resolve, reject) => {
-        try {
+const profile = function (token, userId = "") {
+  return new Promise(async (resolve, reject) => {
+    try {
+      if (env === messageConstants.common.BM) {
```

Comment:
@praveenKDass If the service is running in BM we also need to replace userId in response with the headeruserId which will be coming in userId param

test the survey and observation flows